base/reduce.jl

```diff
@@ -13,20 +13,23 @@ const SmallUnsigned = Union{UInt8,UInt16,UInt32}
 end
 
 const CommonReduceResult = Union{UInt64,UInt128,Int64,Int128,Float32,Float64}
-const WidenReduceResult = Union{SmallSigned, SmallUnsigned, Float16}
+const SmallReduceResult = Union{SmallSigned, SmallUnsigned, Float16}
```

Comment:
If `Float16` is a "real computational type"  https://github.com/JuliaLang/julia/issues/5942#issuecomment-233034260 it shouldn't be widened, right?